gstd-meta/src/interaction.rs

```diff
@@ -0,0 +1,57 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
```

Comment:
what is `interaction.rs` name for?

good practice would be add module-level doc item at least to outline this 
```suggestion
    //! Interaction module ??
```

Reply:
Would it be enough?
```
//! Interaction module provides user the ability to interact with
//! type's metadata and to form json from it
```

Reply:
but did you add it?